src/mrpro/operators/LinearOperatorMatrix.py

```diff
@@ -0,0 +1,357 @@
+        col_numbers = _to_numeric_index(idxs[1], self._shape[1])
+
+        sliced_operators = [
+            [row[col_number] for col_number in col_numbers]
+            for row in [self._operators[row_number] for row_number in row_numbers]
+        ]
+
+        # Return a single operator if only one row and column is selected
+        if len(row_numbers) == 1 and len(col_numbers) == 1:
+            return sliced_operators[0][0]
+        else:
+            return self.__class__(sliced_operators)
+
+    def __iter__(self) -> Iterator[Sequence[LinearOperator]]:
+        """Iterate over the rows of the Operator Matrix."""
+        return iter(self._operators)
+
+    def __repr__(self):
+        """Representation of the Operator Matrix."""
+        return f'LinearOperatorMatrix(shape={self._shape}, operators={self._operators})'
+
+    # Note: The type ignores are needed because we currently cannot do arithmetic operations with non-linear operators.
+    def __add__(self, other: Self | LinearOperator | torch.Tensor) -> Self:  # type: ignore[override]
+        """Addition."""
+        operators: list[list[LinearOperator]] = []
+        if isinstance(other, LinearOperatorMatrix):
+            if self.shape != other.shape:
+                raise ValueError('OperatorMatrix shapes do not match.')
+            for self_row, other_row in zip(self._operators, other._operators, strict=False):
+                operators.append([s + o for s, o in zip(self_row, other_row, strict=False)])
+        elif isinstance(other, LinearOperator | torch.Tensor):
+            if not self.shape[0] == self.shape[1]:
+                raise NotImplementedError('Cannot add a LinearOperator to a non-square OperatorMatrix.')
+            for i, self_row in enumerate(self._operators):
+                operators.append([op + other if i == j else op for j, op in enumerate(self_row)])
+        else:
+            return NotImplemented  # type: ignore[unreachable]
+        return self.__class__(operators)
+
+    def __radd__(self, other: Self | LinearOperator | torch.Tensor) -> Self:
+        """Right addition."""
+        return self.__add__(other)
+
+    def __mul__(self, other: torch.Tensor | Sequence[torch.Tensor | complex] | complex) -> Self:
+        """LinearOperatorMatrix*Tensor multiplication."""
+        if isinstance(other, torch.Tensor | complex | float | int):
+            other_: Sequence[torch.Tensor | complex] = (other,) * self.shape[1]
+        elif len(other) != self.shape[1]:
+            raise ValueError('Other should have the same length as the operator has columns.')
+        else:
+            other_ = other
+        operators = []
+        for row in self._operators:
+            operators.append([op * o for op, o in zip(row, other_, strict=True)])
+        return self.__class__(operators)
+
+    def __rmul__(self, other: torch.Tensor | Sequence[torch.Tensor] | complex) -> Self:
+        """Tensor*LinearOperatorMatrix multiplication."""
```

Comment:
what is this operation supposed to do? say for example x*A for A = [ A1, A2]?

Reply:
A = [ A1, A2]
 x*A = [x*A1, x*A2]

Reply:
I think I still do not understand the purpose... what is x*A in general if A is a linear operator? For A being a matrix ok, but in general?

Reply:
It is not x*A but Lambda*A, A.H*DCF, etc.
It's is just pre and post multiplication.